libs/openFrameworks/graphics/ofImage.cpp

```diff
@@ -209,23 +214,35 @@ static bool loadImage(ofPixels_<PixelType> & pix, const of::filesystem::path& _f
 		return ofLoadImage(pix, ofLoadURL(_fileName.string()).data);
```

Comment:
.string() should be here try/catched (and it is correct tp down convert, as URI/URLs are not wstring-compatible)

Reply:
I've changed now to c_str(), so one more tiny step of fs::path compatible.
ofLoadURL receives a string parameter anyway, and I wonder if fs::path is a good choice for dealing with URLs too.
Anyway it is much more difficult to find problems with foreign characters in URLs than in files.

Reply:
well, URL should indeed be strings and not paths (I may be confused with the PR's) but as I see it currently `ofLoadUrL()` takes a string which is correct:

https://github.com/openframeworks/openFrameworks/blob/6ad07c350b7af1a7974966af7368e5f7027bdb74/libs/openFrameworks/utils/ofURLFileLoader.cpp#L383

the design problem is that `loadImage` takes "something that could be a path or an URL", and tries to extract an URL from the argument, and if it's not an URL (no http scheme) defaults to a file. that logic is flawed: if `loadImage()` is passed a `filesystem::path`, it's undoubtedly a path (no messing with URI); if it is passed a `string` (and no need to support wstring, as URL are not legal wstring), then the URL dance can take effect, and if it's not an URL, upgrade it to a native path.

Reply:
can we have:
` ofImage::loadImage( filesystem::path filePath  ); ` and ` ofImage::loadImage( string urlArg ); ` or would that be ambiguous ? ( for backwards compatibility reasons )